VisualizationBase/src/nodes/ViewItemNode.h

```diff
@@ -58,7 +62,13 @@ class VISUALIZATIONBASE_API ViewItemNode : public Super<UINode>
 		void setSpacingParent(ViewItemNode* spacingParent);
 		ViewItemNode* spacingParent() const;
 
+		virtual QJsonValue toJson() const;
+
 	private:
+		ViewItemNode(const ViewItem* parent);
+
+		const ViewItem* parent_{};
```

Comment:
I changed it back, using the first suggestion.

I think generally we could go in the way of the second solution. It would solve some issues with parents (as the parent of the VViewItemNode is always the ViewItem itself, we would automatically have a pointer to this), and allow us to get rid of some code. Also, by immediately having items and not nodes, probably some of the code which waits until the visualizations actually exist could be removed and perhaps made a bit more efficient.